lib/datadog/opentelemetry/sdk/trace/span.rb

```diff
@@ -15,6 +15,20 @@ def set_attribute(key, value)
           res
         end
 
+        def record_exception(exception, attributes: nil)
+          res = super
+          if (span = datadog_span)
+            span.set_error(
```

Comment:
According to the Datadog specification when an exception is recorded the error status, message, and stacktrace must be set on a span. This is to maintain compatibility with the error tracking product. If multiple exceptions are recorded on a span then the last exception wins.